ledger-service/db-backend/src/main/scala/com/digitalasset/http/dbbackend/Queries.scala

```diff
@@ -162,6 +162,21 @@ sealed abstract class Queries(tablePrefix: String) {
 
   final def surrogateTemplateId(packageId: String, moduleName: String, entityName: String)(implicit
       log: LogHandler
+  ): ConnectionIO[SurrogateTpId] = {
+    tpIdCachedValue(packageId, moduleName, entityName)
+      .map { tpId =>
+        Applicative[ConnectionIO].pure(tpId)
+      }
+      .getOrElse {
+        surrogateTemplateIdDb(packageId, moduleName, entityName).map { tpId =>
+          setTpIdCacheValue(packageId, moduleName, entityName, tpId)
+          tpId
+        }
+      }
+  }
+
+  private def surrogateTemplateIdDb(packageId: String, moduleName: String, entityName: String)(
```

Comment:
need a better name for this function :-/

Reply:
```suggestion
  private def surrogateTemplateIdFromDb(packageId: String, moduleName: String, entityName: String)(
```
I don't know, maybe?